File: GZCTF/ClientApp/src/client/models/Notice.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type Notice = {
    id?: number;
    /**
     * 通知标题
     */
    title: string;
    /**
     * 通知内容
     */
    content: string;
    /**
     * 是否置顶
     */
    isPinned: boolean;
    /**
     * 发布时间
     */
    time: string;
};

File: GZCTF/Models/Data/Notice.cs
using CTFServer.Models.Request.Edit;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace CTFServer.Models;

public class Notice
{
    [Key]
    [Required]
    public int Id { get; set; }

    /// <summary>
    /// 通知标题
    /// </summary>
    [Required]
    public string Title { get; set; } = string.Empty;

    /// <summary>
    /// 通知内容
    /// </summary>
    [Required]
    public string Content { get; set; } = string.Empty;

    /// <summary>
    /// 是否置顶
    /// </summary>
    [Required]
    public bool IsPinned { get; set; } = false;

    /// <summary>
    /// 发布时间
    /// </summary>
    [Required]
    [JsonPropertyName("time")]
    public DateTimeOffset PublishTimeUTC { get; set; } = DateTimeOffset.UtcNow;

    public void UpdateInfo(NoticeModel model)
    {
        Title = model.Title;
        Content = model.Content;
        IsPinned = model.IsPinned;
        PublishTimeUTC = DateTimeOffset.UtcNow;
    }
}